python/ray/experimental/shuffle.py

```diff
@@ -240,19 +256,38 @@ def input_reader(i: PartitionID) -> Iterable[InType]:
     def output_writer(i: PartitionID,
                       shuffle_inputs: List[ObjectRef]) -> OutType:
         total = 0
-        # TODO(ekl) using ray.wait can be more efficient for pipelining.
-        for obj_ref in shuffle_inputs:
-            arr = ray.get(obj_ref)
+        ready, unready = ray.wait(shuffle_inputs)
+        while unready:
+            for obj_ref in ready:
+                arr = ray.get(obj_ref)
+                total += arr.size * arr.itemsize
+            ready, unready = ray.wait(unready)
+            time.sleep(0.1)
```

Comment:
Can you revert this for now? The behavior could be pretty different. We could also make it a flag.

Reply:
Also, why the sleep?

Reply:
You can do this instead:
```
while shuffle_inputs:
    [ready], shuffle_inputs = ray.wait(shuffle_inputs, num_returns=1)
    ray.get(ready)
```

Reply:
oh no reason for sleep. I added for debugging and forgot to remove after I fixed a bug.

Reply:
Added a flag.